docs/reference_guide.md

```diff
@@ -2369,6 +2371,32 @@ ioctl(PERF_EVENT_IOC_SET_BPF): Invalid argument
 Error attaching probe: 'kprobe:vfs_read'
 ```
 
+## 21. `buf()`: Buffers
+
+Syntax: `buf(void *d, int length)`
```

Comment:
Maybe document how the lengths can be inferred on some types. I think in the future with more distros shipping BTF support we can start being able to infer more. So it'll be a nice over-time improvement.